### PR TITLE
cmd: add warning message when grpc proxy address env exists

### PR DESCRIPTION
**Pull request by db-will**
### What problem does this PR solve? 
#1482

### What is changed and how it works?
Add warning log to dm-master, dm-worker and dm-ctl when grpc proxy address environment variables are set(HTTP_PROXY, HTTPS_PROXY, NO_PROXY)

### Check List 

Tests 

 - Unit test: pkg/utils/util_test.go
 - Integration test: tests/http_proxies

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note

### CODE REVIEW

File: tests/http_proxies/run.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	# check dm master log
+	check_log_contains "$WORK_DIR/master/log/dm-master.log" $expected_str 1
+
+	# rum dm worker
+	run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $cur/conf/dm-worker1.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER1_PORT
+
+	# check dm worker log
+	check_log_contains "$WORK_DIR/worker1/log/dm-worker.log" $expected_str 1
+
```
Comment (lance6716) [inline tests/http_proxies/run.sh:30]: a more troublesome scenario is `dmctl` has set a wrong proxy environment, so user can't control the DM cluster. we might also check `dmctl` here.

and maybe it's hard to find the log file of `dmctl`, we can also print the warning message to stdout as an alternative.

**Review by lance6716 — COMMENTED** [review]
new tests should add a line of its name in `others_integration.txt`, because we run test based on some hard-coded names in CI script and content of this txt file.

File: tests/http_proxies/run.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	unset $env_name
+}
+
+function run() {
+	echo "test dm grpc proxy env setting checking for http_proxy=http://127.0.0.1:8080"
+	test_dm "http_proxy" "http://127.0.0.1:8080"
```
Comment (okJiang) [inline tests/http_proxies/run.sh:36]: The issue seems to mean that grpc will use the system default environment variables.

So, maybe you can obtain them and input them here?

Comment (lance6716) [reply tests/http_proxies/run.sh:36]: he set environment variables in this test 🤔

Comment (okJiang) [reply tests/http_proxies/run.sh:36]: Sorry, I didn't see it just now....

File: pkg/utils/util_test.go [hunk 3]
```diff
@@ -197,3 +198,34 @@ func (t *testUtilsSuite) TestIsBuildInSkipDDL(c *C) {
...
 	c.Assert(IsBuildInSkipDDL("DROP PROCEDURE"), IsTrue)
 }
+
+func (t *testUtilsSuite) TestProxyFields(c *C) {
+	revIndex := map[string]int{
+		"http_proxy":  0,
+		"https_proxy": 1,
+		"no_proxy":    2,
+	}
+	envs := [...]string{"http_proxy", "https_proxy", "no_proxy"}
```
[the remark this answers is not in the document]
Comment (db-will) [reply pkg/utils/util_test.go:208]: `[...]` is short for creating array without specifying len of array.

File: tests/http_proxies/run.sh (hunk 1 above)
Comment (db-will) [reply tests/http_proxies/run.sh:30]: the log file is actually hardcoded in `cmd/dm-ctl/main.go` as `/dmctl.log`

File: pkg/utils/util_test.go [hunk 4]
```diff
@@ -197,3 +198,34 @@ func (t *testUtilsSuite) TestIsBuildInSkipDDL(c *C) {
...
+		for i := 0; i < 3; i++ {
+			if (1<<i)&mask != 0 {
+				c.Assert(os.Setenv(envs[i], envPreset[i]), IsNil)
+			}
+		}
+
+		for _, field := range proxyFields() {
+			idx, ok := revIndex[field.Key]
+			c.Assert(ok, IsTrue)
+			c.Assert((1<<idx)&mask, Not(Equals), 0)
```
[the remark this answers is not in the document]
Comment (lance6716) [reply pkg/utils/util_test.go:227]: I have checked that they are both `int` type. It's OK.

**Review by lance6716 — APPROVED** [review]

File: pkg/utils/util.go [hunk 5]
```diff
@@ -28,7 +28,11 @@ import (
 	"github.com/pingcap/tidb/errno"
 
 	"github.com/pingcap/dm/dm/pb"
+	"github.com/pingcap/dm/pkg/log"
 	"github.com/pingcap/dm/pkg/terror"
+
+	"go.uber.org/zap"
+	"golang.org/x/net/http/httpproxy"
```
Comment (lichunzhu) [inline pkg/utils/util.go:35]: merge these with L24~28?

**Review by lichunzhu — COMMENTED** [review]
Rest LGTM
Should we print warning message to stdout for `dmctl`?

**Review by glorv — COMMENTED** [review]
LGTM after remaining comments are resolves

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**